test/model/evaluation.jl

```diff
@@ -1,4 +1,37 @@
-@testset "controlling sampling behavior for conditioned variables" begin
+function _logjoint(model::JuliaBUGS.BUGSModel)
+    _, log_densities = JuliaBUGS.evaluate_with_env!!(model; transformed=model.transformed)
+    return log_densities.tempered_logjoint
+end
+
+function verify_log_densities_structure(log_densities)
+    @test isa(log_densities, NamedTuple)
+    @test haskey(log_densities, :logprior)
+    @test haskey(log_densities, :loglikelihood)
+    @test haskey(log_densities, :tempered_logjoint)
+end
+
+function test_bugs_model_log_density(
+    model_example, expected_untransformed, expected_transformed=expected_untransformed
+)
+    (; model_def, data, inits) = model_example
+    transformed_model = compile(model_def, data, inits)
+    untransformed_model = JuliaBUGS.settrans(transformed_model, false)
+
+    # Allow world age to advance by calling the functions in a separate evaluation
+    Base.invokelatest() do
```

Comment:
Why is this needed? Asking just because I want to learn to understand this world age stuff.

Reply:
In JuliaBUGS, I take the AST of the model definition and generate a function for each variable. These function are basically the RHS in DynamicPPL. 

I then call individual functions as I walk through graph nodes.

I create these functions by `eval`, so there are world-age problems. Hence I need to use `invokelatest`